firebase-firestore/src/main/java/com/google/firebase/firestore/util/CustomClassMapper.java

```diff
@@ -97,6 +98,15 @@ public static Object convertToPlainJavaTypes(Object object) {
     return deserializeToClass(object, clazz, ErrorPath.EMPTY);
   }
 
+  /**
+   * If there are fields annotated with {@link com.google.firebase.firestore.DocumentId}, make sure
+   * they are set to expected value by throwing runtime exception when it fails to comply.
+   *
+   * @param object Java object that might have DocumentId annotated fields.
+   * @param expected Expected value of the DocumentId annotated fields.
+   */
+  public static void assertAnnotatedDocumentIdEqual(Object object, @Nullable Object expected) {}
```

Comment:
As I mentioned elsewhere, we will always allow either null or the expected value, so `expected` won't need to be `@Nullable`, but the implementation should always accept null as well.

Reply:
No longer needed.